rekall-core/rekall/plugins/linux/heap_analysis.py

```diff
@@ -1787,20 +1797,24 @@ def _initialize_heap_vma_list(self):
             if vma['name'] == self._main_heap_identifier:
                 self.heap_vmas.append(vma)
 
-        for arena in self.arenas:
-            if arena.is_main_arena:
-                for mmap_chunk in arena.mmapped_first_chunks:
-                    vma = get_vma_for_offset(self.vmas, mmap_chunk.v())
-                    if vma not in self.heap_vmas:
-                        self.heap_vmas.append(vma)
+        for mmap_chunk in self.get_all_mmapped_chunks():
+            self._add_mmapped_chunk_to_heap_vma_list(mmap_chunk)
 
-            else:
+        for arena in self.arenas:
+            if not arena.is_main_arena:
                 for heap in arena.heaps:
                     vma = get_vma_for_offset(self.vmas, heap.v())
                     if vma not in self.heap_vmas:
                         self.heap_vmas.append(vma)
 
 
+    def _add_mmapped_chunk_to_heap_vma_list(self, mmapped_chunk):
+        memr = Memregion(mmapped_chunk.v(), mmapped_chunk.v() + mmapped_chunk.chunksize())
+        vma = {'name': 'mmapped_chunk', 'vma': memr}
+        if vma['vma'].vm_start not in [x['vma'].vm_start for x in self.heap_vmas]:
```

Comment:
It might be worth keeping a second set() object here - recalculating this each time is O(n^2)